### PR TITLE
Docs/userEndpoints: updated documentation for user endpoints

### PR DESCRIPTION
**Pull request by inuoluwadunsimi**

### CODE REVIEW

File: spec/api.yaml [hunk 1]
```diff
@@ -124,6 +140,39 @@ paths:
               schema:
                 $ref: '#/components/schemas/User'
 
+  /user/edit-profile:
```
Comment (Lekky71) [inline spec/api.yaml]: Endpoint can just be called `/user/profile`, semantically we know edit request by the PUT method

File: spec/api.yaml [hunk 2]
```diff
@@ -124,6 +140,39 @@ paths:
...
+       put:
+         summary: 'edit user details'
+         tags:
+           -user
+         parameters:
+           -$ref: '#/components/parameters/AuthToken'
```
Comment (Lekky71) [inline spec/api.yaml]: Put spaces after your -

File: spec/api.yaml [hunk 3]
```diff
@@ -124,6 +140,39 @@ paths:
...
+         operationId: EditUserProfile
+         requestBody: 
+            content:
+             application/json:
+               schema:
+                 $ref: '#/components/schemas/SignUpUserInfoRequest'
```
Comment (Lekky71) [inline spec/api.yaml]: Don't use `SignUpUserInfoRequest` as it contains password, create a new schema that only takes in the name that the user wants to change to.

File: spec/api.yaml [hunk 4]
```diff
@@ -124,6 +140,39 @@ paths:
...
+             application/json:
+               schema:
+                 $ref: '#/components/schemas/SignUpUserInfoRequest' 
+         responses:
+           200:
+            description: 'Success'
+            content:
+              application/json:
+                schema:
+                $ref: '#components/schemas/User'
```
Comment (Lekky71) [inline spec/api.yaml]: Indent

File: spec/api.yaml [hunk 5]
```diff
@@ -124,6 +140,39 @@ paths:
...
+           400:
```
Comment (Lekky71) [inline spec/api.yaml]: Not Found is error 404.

File: spec/api.yaml [hunk 6]
```diff
@@ -108,6 +111,19 @@ paths:
                     type: string
                 example:
                   { "message": "Invalid Login details" }
+        400:
```
Comment (Lekky71) [inline spec/api.yaml]: Make this an error 401 Unauthorized instead. Why are you returning the OTP? It is something you send via email, not in the response.

File: spec/api.yaml [hunk 7]
```diff
@@ -97,7 +100,7 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/User'
-        400:
+        401:
```
Comment (Lekky71) [inline spec/api.yaml]: Revert this to 400 BadRequest

File: spec/api.yaml [hunk 8]
```diff
@@ -85,6 +85,9 @@ paths:
         - onboarding
       operationId: LoginUser
       summary: 'Login User'
+      parameters:
+         - name: deviceId
```
Comment (Lekky71) [inline spec/api.yaml]: Let the name be x-device-id, and define it in the components.parameters instead, so it can be re-used.

**Review by Lekky71 — CHANGES_REQUESTED** [review]
Good work, a number of changes to be made and you should also test the docs by running the server and then go to /docs in your browser.

File: spec/api.yaml (hunk 8 above)
Comment (inuoluwadunsimi) [reply spec/api.yaml]: > Good work, a number of changes to be made and you should also test the docs by running the server and then go to /docs in your browser.

I have made the changes now but I am having an issue running it on my computer  " example-app@1.0.0 dev:start", This is the error it is showing.

File: spec/api.yaml [hunk 9]
```diff
@@ -124,6 +137,39 @@ paths:
               schema:
                 $ref: '#/components/schemas/User'
 
+  /user/profile:
+       put:
+         summary: 'edit user details'
+         tags:
+           -user
```
Comment (Lekky71) [inline spec/api.yaml]: Indent

**Review by Lekky71 — CHANGES_REQUESTED** [review]

**Review by Lekky71 — APPROVED** [review]
Awesome

**Pull request merged**